src/server.ts

```diff
@@ -0,0 +1,98 @@
+import 'dotenv/config';
+
+import cors from '@fastify/cors';
+import Fastify from 'fastify';
+import type {Feature, FeatureCollection, Geometry} from 'geojson';
+import {addElevation} from './elevation/addElevation.js';
+
+const port = parseInt(process.env.PORT!);
+const connectionTimeout = parseInt(process.env.CONNECTION_TIMEOUT!);
+const keepAliveTimeout = parseInt(process.env.KEEP_ALIVE_TIMEOUT!);
+const bodyLimit = parseInt(process.env.MAX_POST_SIZE!);
+
+const fastify = Fastify({
+    logger: true,
+    ignoreTrailingSlash: true,
+    disableRequestLogging: true,
+    // 500kb
+    bodyLimit,
+    connectionTimeout,
+    keepAliveTimeout,
+});
+
+// Add header for connection timeout to all responses
+fastify.server.headersTimeout = connectionTimeout;
+
+await fastify.register(cors, {
+    origin: (origin, cb) => {
+        switch (origin) {
+            case 'http://localhost:8000':
+            case 'https://www.gaiagps.com':
+            case 'https://www-staging.gaiagps.com':
```

Comment:
I assume we need to add admin too?

```suggestion
            case 'https://www-staging.gaiagps.com':
            case 'https://www-admin.gaiagps.xyz':
```